### DISCUSSION TITLE
Welcome to PyMuPDF-Utilities Discussions!

### DISCUSSION THREAD

**Discussion by JorjMcKie** [pymupdf/PyMuPDF-Utilities#33] [Announcements]
I decided to open a separate Discussions area for this repository.
It seems that enough things are coming up that cover topics addressed here.

**Comment by zhuzhangwei** [comment]
"Hello! I have a question regarding your pymupdf script for changing fonts in PDF. I'm a novice programmer, and I've carefully studied your repl-font.py and repl-fontnames.py scripts. I'd like to inquire about the part of your code that deals with replacing old fonts with new ones. It appears that your code establishes a mapping of old fonts to new fonts, but what if the fonts in the PDF are unknown or not mainstream fonts? For example, fonts like AdvOTce3d9a73, AdvOTce3d9a73, AdvOTd369e91e, AdvOT8608a8d1, AdvOTd369e91e, AdvOTd369e91e, AdvOT8608a8d1, AdvOTd369e91e – in such cases, does your code still apply? Does your code have a provision for setting unknown fonts to a specific font? I hope to receive your guidance. As a newcomer to programming, I would greatly benefit from your insights."

I hope this helps! If you have any more questions or need further assistance, feel free to ask.

**Reply by JorjMcKie** [reply]
Whether the code works (or how good) depends on the new fonts you want to use.
The script does not do any deep research on what old fonts exactly are - the JSON file produced by the first step contains all information needed by the subsequent step.
This also means, that "non-mainstream" fonts can be handled just as well, technically. The only potential problem is for you, because you probably don't know what the heck of a font "AdvOT8608a8d1" is. So you probably have a hard time to decide whether to replace it.

If the latter is your problem, you try and extract text in a way that also shows the used font alongside the text itself: look at the spans produced by `page.get_text("dict")`.

Do not hesitate to ask if you have more questions.

**Comment by JorjMcKie** [comment]
By the way:
In the future, please start a new discussion for whatever you want to ask or contribute.
Do not modify this post.
